### PR TITLE
Beast shape

### PR DESCRIPTION
**Pull request by rdrrichards**

### CODE REVIEW

**Comment by cvillhauer** [comment]
@rdrrichards Ah ha! Thank you so much!!

**Comment by rdrrichards** [comment]
I think I got a little over zealous by creating this PR. Feel free to delete this.

File: src/app/services/spell.service.ts [hunk 1]
```diff
@@ -2,40 +2,81 @@ import { Injectable } from '@angular/core';
...
-            map(spells => {
-                return spells.filter(spell => spell.level <= level);
-            })
-        );
-    }
+  getSpellsByCharacterClassAndLevel(characterClass: string, level: number): Observable<Spell[]> {
+    let spells: Observable<Spell[]>;
+    const spellGroup = this.calculateSpellGroupFromClass(characterClass);
+
+    // TODO: How do I combine these two items into one returnable Observable?
```
Comment (cvillhauer) [inline src/app/services/spell.service.ts]: @rdrrichards I need help on this... I want to make one call to the Spell Service and get back a (observable) list of Spells, some of them Summon spells and some of them Shapeshift spells and some of them regular spells. I've been experimenting with forkjoin and not able to get it to do what I want. I'm probably missing something obvious.

Comment (rdrrichards) [reply src/app/services/spell.service.ts]: @cvillhauer Try this:

```
  getSpellsByCharacterClassAndLevel(characterClass: string, level: number): Observable<Spell[]> {
    const spellGroup = this.calculateSpellGroupFromClass(characterClass);

    // TODO: How do I combine these two items into one returnable Observable?
    return forkJoin([this.getShapeshiftSpellsBySpellLevel(level),
      this.getSummonSpellsBySpellGroupAndLevel(spellGroup, level)]).pipe(
        map(value => [...value[0], ...value[1]])
      );
  }
```

**Comment by cvillhauer** [comment]
@rdrrichards Yep, I'm definitely not ready for a PR on this :-) Long way to go with refactoring and feature-adding. I plan to keep working on it this week!

**Pull request closed without merging**